### PR TITLE
update scipy easyblock for scipy >= 1.9.0 to use meson/ninja 

### PR DESCRIPTION
**Pull request by jfgrimm**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by jfgrimm** [comment]
figuring out the best way to ensure all the tests are run in a way they don't fail silently

**Comment by jfgrimm** [comment]
Test report by @jfgrimm
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2861, https://github.com/easybuilders/easybuild-easyblocks/pull/2862
SUCCESS
Build succeeded for 6 out of 6 (6 easyconfigs in total)
himem06.pri.viking.alces.network - Linux CentOS Linux 7.9.2009, x86_64, Intel(R) Xeon(R) Platinum 8160 CPU @ 2.10GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/d4871a9ec3d7bf99984c4dc547e674e9 for a full test report.

**Comment by jfgrimm** [comment]
Test report by @jfgrimm
#### Overview of tested easyconfigs (in order)
 * **FAIL (unhandled exception: 'python')Traceback (most recent call last):
  File "/opt/apps/easybuild/software/EasyBuild/4.7.0/lib/python3.6/site-packages/easybuild/main.py", line 132, in build_and_install_software
    (ec_res['success'], app_log, err) = build_and_install_one(ec, init_env)
  File "/opt/apps/easybuild/software/EasyBuild/4.7.0/lib/python3.6/site-packages/easybuild/framework/easyblock.py", line 4182, in build_and_install_one
    result = app.run_all_steps(run_test_cases=run_test_cases)
  File "/opt/apps/easybuild/software/EasyBuild/4.7.0/lib/python3.6/site-packages/easybuild/framework/easyblock.py", line 4065, in run_all_steps
    self.run_step(step_name, step_methods)
  File "/opt/apps/easybuild/software/EasyBuild/4.7.0/lib/python3.6/site-packages/easybuild/framework/easyblock.py", line 3900, in run_step
    step_method(self)()
  File "/opt/apps/easybuild/software/EasyBuild/4.7.0/lib/python3.6/site-packages/easybuild/easyblocks/generic/pythonbundle.py", line 129, in extensions_step
    super(PythonBundle, self).extensions_step(*args, **kwargs)
  File "/opt/apps/easybuild/software/EasyBuild/4.7.0/lib/python3.6/site-packages/easybuild/framework/easyblock.py", line 2845, in extensions_step
    self.install_extensions(install=install)
  File "/opt/apps/easybuild/software/EasyBuild/4.7.0/lib/python3.6/site-packages/easybuild/framework/easyblock.py", line 1838, in install_extensions
    self.install_extensions_sequential(install=install)
  File "/opt/apps/easybuild/software/EasyBuild/4.7.0/lib/python3.6/site-packages/easybuild/framework/easyblock.py", line 1886, in install_extensions_sequential
    txt = ext.run()
  File "/opt/apps/easybuild/software/EasyBuild/4.7.0/lib/python3.6/site-packages/easybuild/easyblocks/generic/pythonpackage.py", line 772, in run
    step_method(self)()
  File "/opt/apps/easybuild/software/EasyBuild/4.7.0/lib/python3.6/site-packages/easybuild/framework/easyblock.py", line 2682, in _test_step
    self.test_step()
  File "/tmp/eb-hhd5rp_9/included-easyblocks-n24bwvxu/easybuild/easyblocks/scipy.py", line 160, in test_step
    'parallel': self.cfg['parallel']
KeyError: 'python'
** _SciPy-bundle-2022.05-foss-2022a.eb_ 
 * **FAIL (unhandled exception: 'python')Traceback (most recent call last):
  File "/opt/apps/easybuild/software/EasyBuild/4.7.0/lib/python3.6/site-packages/easybuild/main.py", line 132, in build_and_install_software
    (ec_res['success'], app_log, err) = build_and_install_one(ec, init_env)
  File "/opt/apps/easybuild/software/EasyBuild/4.7.0/lib/python3.6/site-packages/easybuild/framework/easyblock.py", line 4182, in build_and_install_one
    result = app.run_all_steps(run_test_cases=run_test_cases)
  File "/opt/apps/easybuild/software/EasyBuild/4.7.0/lib/python3.6/site-packages/easybuild/framework/easyblock.py", line 4065, in run_all_steps
    self.run_step(step_name, step_methods)
  File "/opt/apps/easybuild/software/EasyBuild/4.7.0/lib/python3.6/site-packages/easybuild/framework/easyblock.py", line 3900, in run_step
    step_method(self)()
  File "/opt/apps/easybuild/software/EasyBuild/4.7.0/lib/python3.6/site-packages/easybuild/easyblocks/generic/pythonbundle.py", line 129, in extensions_step
    super(PythonBundle, self).extensions_step(*args, **kwargs)
  File "/opt/apps/easybuild/software/EasyBuild/4.7.0/lib/python3.6/site-packages/easybuild/framework/easyblock.py", line 2845, in extensions_step
    self.install_extensions(install=install)
  File "/opt/apps/easybuild/software/EasyBuild/4.7.0/lib/python3.6/site-packages/easybuild/framework/easyblock.py", line 1838, in install_extensions
    self.install_extensions_sequential(install=install)
  File "/opt/apps/easybuild/software/EasyBuild/4.7.0/lib/python3.6/site-packages/easybuild/framework/easyblock.py", line 1886, in install_extensions_sequential
    txt = ext.run()
  File "/opt/apps/easybuild/software/EasyBuild/4.7.0/lib/python3.6/site-packages/easybuild/easyblocks/generic/pythonpackage.py", line 772, in run
    step_method(self)()
  File "/opt/apps/easybuild/software/EasyBuild/4.7.0/lib/python3.6/site-packages/easybuild/framework/easyblock.py", line 2682, in _test_step
    self.test_step()
  File "/tmp/eb-hhd5rp_9/included-easyblocks-n24bwvxu/easybuild/easyblocks/scipy.py", line 160, in test_step
    'parallel': self.cfg['parallel']
KeyError: 'python'
** _SciPy-bundle-2021.05-foss-2021a.eb_ 

Build succeeded for 0 out of 2 (2 easyconfigs in total)
himem06.pri.viking.alces.network - Linux CentOS Linux 7.9.2009, x86_64, Intel(R) Xeon(R) Platinum 8160 CPU @ 2.10GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/43e66f71ca7ef3168fedcc8d072d5b5b for a full test report.

**Comment by jfgrimm** [comment]
Test report by @jfgrimm
#### Overview of tested easyconfigs (in order)
 * **FAIL (build issue)** _SciPy-bundle-2022.05-foss-2022a.eb_ (partial log available at https://gist.github.com/1e808607f23ba070d342a0a72428b662)
 * **FAIL (build issue)** _SciPy-bundle-2021.05-foss-2021a.eb_ (partial log available at https://gist.github.com/bfbfa7a6fa8b83b6707b9acb6c2cd26c)

Build succeeded for 0 out of 2 (2 easyconfigs in total)
himem06.pri.viking.alces.network - Linux CentOS Linux 7.9.2009, x86_64, Intel(R) Xeon(R) Platinum 8160 CPU @ 2.10GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/09121aae431fc0f1327daf30c8066f1a for a full test report.

**Comment by jfgrimm** [comment]
test failures actually are present in old builds, they were just ignored (see https://github.com/easybuilders/easybuild-easyblocks/issues/2237)

**Comment by jfgrimm** [comment]
Test report by @jfgrimm
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _SciPy-bundle-2022.05-foss-2022a.eb_ 
 * **SUCCESS** _SciPy-bundle-2021.05-foss-2021a.eb_ 

Build succeeded for 2 out of 2 (2 easyconfigs in total)
himem06.pri.viking.alces.network - Linux CentOS Linux 7.9.2009, x86_64, Intel(R) Xeon(R) Platinum 8160 CPU @ 2.10GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/7e697701e89c433ca89242d6515ea421 for a full test report.

File: easybuild/easyblocks/s/scipy.py [hunk 1]
```diff
@@ -30,34 +30,166 @@
...
+                'meson configure --prefix=%s' % tmpdir,
+                'ninja -j %s install' % self.cfg['parallel'],
+            ])
+            run_cmd(cmd, log_all=True, simple=True, verbose=False)
+
+            abs_pylibdirs = [os.path.join(tmpdir, pylibdir) for pylibdir in self.all_pylibdirs]
+            for pylibdir in abs_pylibdirs:
+                mkdir(pylibdir, parents=True)
+            python_root = get_software_root('Python')
+            python_bin = os.path.join(python_root, 'bin', 'python')
```
Comment (boegel) [inline easybuild/easyblocks/s/scipy.py:148]: should use `self.python_cmd` here, which can be set by `self.prepare_python()` (defined in `PythonPackage`)

File: easybuild/easyblocks/s/scipy.py [hunk 2]
```diff
@@ -30,34 +30,166 @@
...
+            cmd = " && ".join([
+                # reconfigure meson to use temp dir
+                'meson configure --prefix=%s' % tmpdir,
+                'ninja -j %s install' % self.cfg['parallel'],
+            ])
+            run_cmd(cmd, log_all=True, simple=True, verbose=False)
+
+            abs_pylibdirs = [os.path.join(tmpdir, pylibdir) for pylibdir in self.all_pylibdirs]
+            for pylibdir in abs_pylibdirs:
+                mkdir(pylibdir, parents=True)
```
Comment (boegel) [inline easybuild/easyblocks/s/scipy.py:146]: the directories that don't exist yet are irrelevant for running the tests, so why create them? 🤔

File: easybuild/easyblocks/s/scipy.py [hunk 3]
```diff
@@ -30,34 +30,166 @@
...
+
+    def install_step(self):
+        """Custom install step for scipy: use ninja for scipy >= 1.9.0"""
+        if self.use_meson:
+            # ignore installopts inherited from FortranPythonPackage
+            self.cfg['installopts'] = ""
+            MesonNinja.install_step(self)
+
+        else:
+            super(EB_scipy, self).install_step()
```
Comment (boegel) [inline easybuild/easyblocks/s/scipy.py:191]:
```suggestion
            FortranPythonPackage.install_step(self)

File: easybuild/easyblocks/s/scipy.py [hunk 4]
```diff
@@ -30,34 +30,166 @@
...
+                'srcdir': self.cfg['start_dir'],
+                'parallel': self.cfg['parallel'],
+            }
+            super(EB_scipy, self).test_step()
+
+    def install_step(self):
+        """Custom install step for scipy: use ninja for scipy >= 1.9.0"""
+        if self.use_meson:
+            # ignore installopts inherited from FortranPythonPackage
+            self.cfg['installopts'] = ""
```
Comment (boegel) [inline easybuild/easyblocks/s/scipy.py:187]: This should be done in `__init__`:

```python
        if self.use_meson:
            # avoid that installopts inherited from (Fortran)PythonPackage are used by MesonNinja.install_step
            self.cfg['installopts'] = ''

**Review by boegel — CHANGES_REQUESTED** [review]

**Comment by boegel** [comment]
Test report by @boegel
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _SciPy-bundle-2020.11-foss-2020b.eb_ 
 * **SUCCESS** _SciPy-bundle-2021.05-foss-2021a.eb_ 
 * **SUCCESS** _SciPy-bundle-2021.05-intel-2021a.eb_ 
 * **SUCCESS** _SciPy-bundle-2021.10-foss-2021b.eb_ 
 * **SUCCESS** _SciPy-bundle-2021.10-foss-2021b-Python-2.7.18.eb_ 
 * **SUCCESS** _SciPy-bundle-2021.10-intel-2021b.eb_ 
 * **SUCCESS** _SciPy-bundle-2022.05-foss-2022a.eb_ 
 * **SUCCESS** _SciPy-bundle-2022.05-intel-2022a.eb_ 

Build succeeded for 8 out of 8 (8 easyconfigs in total)
node3102.skitty.os - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/b5dd170a8e3d16664a9a10723b1a6411 for a full test report.

File: easybuild/easyblocks/s/scipy.py [hunk 5]
```diff
@@ -30,34 +30,166 @@
...
+
+            # reconfigure meson to use real install dir
+            change_dir(cwd)
+            cmd = 'meson configure --prefix=%s' % self.installdir
+            run_cmd(cmd, log_all=True, simple=True, verbose=False)
+
+            try:
+                remove_dir(tmpdir)
+            except OSError as err:
+                raise EasyBuildError("Failed to remove directory %s: %s", tmpdir, err)
```
[the remark this answers is not in the document]
Comment (jfgrimm) [reply easybuild/easyblocks/s/scipy.py]: removed

File: easybuild/easyblocks/s/scipy.py [hunk 6]
```diff
@@ -30,34 +30,166 @@
...
+            ])
+            self.cfg['runtest'] = self.testcmd % {
+                'python': python_bin,
+                'srcdir': self.cfg['start_dir'],
+                'parallel': self.cfg['parallel'],
+            }
+
+            MesonNinja.test_step(self)
+
+            # reconfigure meson to use real install dir
```
[the remark this answers is not in the document]
Comment (jfgrimm) [reply easybuild/easyblocks/s/scipy.py]: sounds sensible

File: easybuild/easyblocks/s/scipy.py [hunk 7]
```diff
@@ -30,34 +30,166 @@
...
+        if self.use_meson:
+
+            # temporarily install scipy so we can run the test suite
+            tmpdir = tempfile.mkdtemp()
+            cwd = os.getcwd()
+
+            cmd = " && ".join([
+                # reconfigure meson to use temp dir
+                'meson configure --prefix=%s' % tmpdir,
+                'ninja -j %s install' % self.cfg['parallel'],
```
[the remark this answers is not in the document]
Comment (jfgrimm) [reply easybuild/easyblocks/s/scipy.py]: ah yes, I'd planned to go back and do that once I had it working

File: easybuild/easyblocks/s/scipy.py (hunk 4 above)
Comment (jfgrimm) [reply easybuild/easyblocks/s/scipy.py]: done

File: easybuild/easyblocks/s/scipy.py [hunk 8]
```diff
@@ -30,34 +30,164 @@
...
+class EB_scipy(FortranPythonPackage, PythonPackage, MesonNinja):
     """Support for installing the scipy Python package as part of a Python installation."""
 
+    @staticmethod
+    def extra_options():
+        """Easyconfig parameters specific to scipy."""
+        extra_vars = ({
+            'enable_slow_tests': [False, "Run scipy test suite, including tests marked as slow", CUSTOM],
+            # ignore tests by default to maintain behaviour in older easyconfigs
+            'ignore_test_result': [None, "Run scipy test suite, but ignore result (only log)", CUSTOM],
```
Comment (boegel) [inline easybuild/easyblocks/s/scipy.py:58]: help message should mention what `None` implies (and comment should also be tweaked)

File: easybuild/easyblocks/s/scipy.py [hunk 9]
```diff
@@ -30,34 +30,164 @@
...
+        # calling PythonPackage __init__ also lets MesonNinja work in an extension
+        PythonPackage.__init__(self, *args, **kwargs)
         self.testinstall = True
-        self.testcmd = "cd .. && %(python)s -c 'import numpy; import scipy; scipy.test(verbose=2)'"
+
+        if LooseVersion(self.version) >= LooseVersion('1.9'):
+            self.use_meson = True
+            # enforce scipy test suite results if not explicitly disabled for scipy >= 1.9
+            if self.cfg['ignore_test_result'] is None:
+                self.cfg['ignore_test_result'] = True
```
Comment (boegel) [inline easybuild/easyblocks/s/scipy.py:73]: `enforce` in comment and setting `ignore_test_result` looks (very) contradictory to me...

Shouldn't this be using `False`?

File: easybuild/easyblocks/s/scipy.py [hunk 10]
```diff
@@ -30,34 +30,164 @@
...
+            # ignore installopts inherited from FortranPythonPackage
+            self.cfg['installopts'] = ""
```
Comment (boegel) [inline easybuild/easyblocks/s/scipy.py:75]: We should at least also log the original `installopts` value if we're going to do a hard reset here...

File: easybuild/easyblocks/s/scipy.py [hunk 11]
```diff
@@ -30,34 +30,164 @@
...
+        PythonPackage.__init__(self, *args, **kwargs)
         self.testinstall = True
-        self.testcmd = "cd .. && %(python)s -c 'import numpy; import scipy; scipy.test(verbose=2)'"
+
+        if LooseVersion(self.version) >= LooseVersion('1.9'):
+            self.use_meson = True
+            # enforce scipy test suite results if not explicitly disabled for scipy >= 1.9
+            if self.cfg['ignore_test_result'] is None:
+                self.cfg['ignore_test_result'] = True
+            # ignore installopts inherited from FortranPythonPackage
```
Comment (boegel) [inline easybuild/easyblocks/s/scipy.py:74]: `FortranPythonPackage` -> `PythonPackage`

File: easybuild/easyblocks/s/scipy.py [hunk 12]
```diff
@@ -30,34 +30,164 @@
...
+            self.cfg['installopts'] = ""
+        else:
+            self.use_meson = False
+
+        if self.cfg['ignore_test_result']:
+            # maintains compatibility with easyconfigs predating scipy 1.9. Runs tests (serially) in
```
Comment (boegel) [inline easybuild/easyblocks/s/scipy.py:80]: `maintains` -> `can be used to maintain`

File: easybuild/easyblocks/s/scipy.py [hunk 13]
```diff
@@ -30,34 +30,164 @@
...
+            if lapack_lib == toolchain.FLEXIBLAS:
+                blas_lapack = 'flexiblas'
+            elif lapack_lib == toolchain.INTELMKL:
+                blas_lapack = 'mkl'
+            elif lapack_lib == toolchain.OPENBLAS:
+                blas_lapack = 'openblas'
+            else:
+                raise EasyBuildError("Unknown BLAS/LAPACK library used: %s", lapack_lib)
+
+            configopts = '-Dblas=%(blas_lapack)s -Dlapack=%(blas_lapack)s' % {'blas_lapack': blas_lapack}
```
Comment (boegel) [inline easybuild/easyblocks/s/scipy.py:109]: maybe rename `configopts` to `blas_lapack_config_opts`?

Comment (boegel) [reply easybuild/easyblocks/s/scipy.py:109]: and maybe this is cleaner/better, not sure:

```python
for opt in ('blas', 'lapack'):
    self.cfg.update('-D' + opt + '=' + blas_lapack)

File: easybuild/easyblocks/s/scipy.py [hunk 14]
```diff
@@ -30,34 +30,164 @@
...
+            pythonpath = os.getenv('PYTHONPATH')
+            pylibdir = det_pylibdir()
+            env.setvar('PYTHONPATH', os.pathsep.join([os.path.join(self.installdir, pylibdir), pythonpath]))
+
+            path = os.getenv('PATH')
+            env.setvar('PATH', os.pathsep.join([os.path.join(self.installdir, 'bin'), path]))
+
+            MesonNinja.configure_step(self)
+
+        else:
```
Comment (boegel) [inline easybuild/easyblocks/s/scipy.py:122]: maybe also add a comment here: `# scipy < 1.9.0 uses install procedure using setup.py`

File: easybuild/easyblocks/s/scipy.py [hunk 15]
```diff
@@ -30,34 +30,164 @@
...
+                'srcdir': self.cfg['start_dir'],
+                'parallel': self.cfg['parallel'],
+            }
+            FortranPythonPackage.test_step(self)
+
+    def install_step(self):
+        """Custom install step for scipy: use ninja for scipy >= 1.9.0"""
+        if self.use_meson:
+            MesonNinja.install_step(self)
+
```
Comment (boegel) [inline easybuild/easyblocks/s/scipy.py:187]: kill this blank line?

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/easyblocks/s/scipy.py (hunk 9 above)
Comment (jfgrimm) [reply easybuild/easyblocks/s/scipy.py:73]: indeed, that should be `False`

File: easybuild/easyblocks/s/scipy.py (hunk 14 above)
Comment (jfgrimm) [reply easybuild/easyblocks/s/scipy.py:133]: done

File: easybuild/easyblocks/s/scipy.py (hunk 10 above)
Comment (jfgrimm) [reply easybuild/easyblocks/s/scipy.py]: updated to strip the ones inherited from PythonPackage, perhaps that's better?

**Comment by branfosj** [comment]
```
== 2023-02-16 11:27:10,457 build_log.py:171 ERROR EasyBuild crashed with an error (at easybuild/src/easybuild-framework/easybuild/base/exceptions.py:126 in __init__): cmd " unset LDFLAGS &&   ninja -j 20  --no-deps  --ignore-installed  --no-index  install" exited with exit code 1 and output:
ninja: unrecognized option '--no-deps'
usage: ninja [options] [targets...]

if targets are unspecified, builds the 'default' target (see manual).

options:
  --version      print ninja version ("1.11.1")
  -v, --verbose  show all command lines while building
  --quiet        don't show progress status, just command output

  -C DIR   change to DIR before doing anything else
  -f FILE  specify input build file [default=build.ninja]

  -j N     run N jobs in parallel (0 means infinity) [default=20 on this system]
  -k N     keep going until N jobs fail (0 means infinity) [default=1]
  -l N     do not start new jobs if the load average is greater than N
  -n       dry run (don't run commands but act like they succeeded)

  -d MODE  enable debugging (use '-d list' to list modes)
  -t TOOL  run a subtool (use '-t list' to list subtools)
    terminates toplevel options; further flags are passed to the tool
  -w FLAG  adjust warnings (use '-w list' to list warnings)
```

From a build of https://github.com/easybuilders/easybuild-easyconfigs/pull/16912 started at 11:19:59 - so with 016118a

File: easybuild/easyblocks/s/scipy.py [hunk 16]
```diff
@@ -30,34 +30,175 @@
...
+                self.cfg['ignore_test_result'] = False
+
+            # strip inherited PythonPackage installopts
+            installopts = self.cfg['installopts']
+            pythonpackage_installopts = ['--no-deps', '--ignore-installed', '--no-index', '--egg',
+                                         '--zip-ok', '--no-index']
+            self.log.info("Stripping inherited PythonPackage installopts %s from installopts %s",
+                          pythonpackage_installopts, installopts)
+            for i in pythonpackage_installopts:
+                installopts.replace(i, '')
```
Comment (branfosj) [inline easybuild/easyblocks/s/scipy.py:84]:
```suggestion
                installopts = installopts.replace(i, '')
```

Comment (jfgrimm) [reply easybuild/easyblocks/s/scipy.py]: whoops

**Comment by boegel** [comment]
Test report by @boegel
#### Overview of tested easyconfigs (in order)
 * **FAIL (build issue)** _SciPy-bundle-2019.10-foss-2019b-Python-3.7.4.eb_ (partial log available at https://gist.github.com/01e2d8bbb8dabbfeac83cb500ae1461d)
 * **FAIL (build issue)** _SciPy-bundle-2019.10-fosscuda-2019b-Python-3.7.4.eb_ (partial log available at https://gist.github.com/cce76a2554d2cb3fd5fb33af53c09459)
 * **FAIL (build issue)** _SciPy-bundle-2020.11-foss-2020b.eb_ (partial log available at https://gist.github.com/44f873d3753cd947c66dac10e80b6857)
 * **FAIL (build issue)** _SciPy-bundle-2020.11-fosscuda-2020b.eb_ (partial log available at https://gist.github.com/b3b273550d42dc995e63cae047dd402a)
 * **FAIL (build issue)** _SciPy-bundle-2021.05-foss-2021a.eb_ (partial log available at https://gist.github.com/78261875bf9911e29e1c7be606b73fbf)
 * **FAIL (build issue)** _SciPy-bundle-2021.05-intel-2021a.eb_ (partial log available at https://gist.github.com/7b8eac2e4c7b7d2043f2561c9b5e15a2)
 * **FAIL (build issue)** _SciPy-bundle-2021.10-foss-2021b.eb_ (partial log available at https://gist.github.com/f5ea7e4c4f38aabed33ad20aefd68258)
 * **FAIL (build issue)** _SciPy-bundle-2021.10-foss-2021b-Python-2.7.18.eb_ (partial log available at https://gist.github.com/00cf2f247ba58f21c81052f3e1c24276)
 * **FAIL (build issue)** _SciPy-bundle-2021.10-intel-2021b.eb_ (partial log available at https://gist.github.com/3584dda423e7cb1ff10b6733cabccf3b)
 * **FAIL (build issue)** _SciPy-bundle-2022.05-foss-2022a.eb_ (partial log available at https://gist.github.com/71d9e61c150b3a3d496662a0328d4c11)
 * **FAIL (build issue)** _SciPy-bundle-2022.05-intel-2022a.eb_ (partial log available at https://gist.github.com/caa9210b983ba30e7d4563f29b7564fd)

Build succeeded for 0 out of 11 (11 easyconfigs in total)
node3125.skitty.os - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/573af2e39d236ba448ee54ad9934e6d6 for a full test report.

File: easybuild/easyblocks/s/scipy.py [hunk 17]
```diff
@@ -30,34 +30,179 @@
...
+
+        else:
+            self.use_meson = False
+
+        if self.cfg['ignore_test_result'] is None:
+            # enforce scipy test suite results if not explicitly disabled for scipy >= 1.9
+            if self.use_meson:
+                self.cfg['ignore_test_result'] = False
+            else:
+                self.cfg['ignore_test_result'] = True
```
Comment (boegel) [inline easybuild/easyblocks/s/scipy.py:92]: @jfgrimm Let's simplify this a bit, see https://github.com/jfgrimm/easybuild-easyblocks/pull/2

**Review by boegel — CHANGES_REQUESTED** [review]

**Comment by jfgrimm** [comment]
Test report by @jfgrimm
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _SciPy-bundle-2019.10-foss-2019b-Python-3.7.4.eb_ 
 * **SUCCESS** _SciPy-bundle-2019.10-fosscuda-2019b-Python-3.7.4.eb_ 
 * **SUCCESS** _SciPy-bundle-2020.11-foss-2020b.eb_ 
 * **SUCCESS** _SciPy-bundle-2020.11-fosscuda-2020b.eb_ 
 * **SUCCESS** _SciPy-bundle-2021.05-foss-2021a.eb_ 
 * **SUCCESS** _SciPy-bundle-2021.05-intel-2021a.eb_ 
 * **SUCCESS** _SciPy-bundle-2021.10-foss-2021b.eb_ 
 * **SUCCESS** _SciPy-bundle-2021.10-intel-2021b.eb_ 
 * **SUCCESS** _SciPy-bundle-2022.05-foss-2022a.eb_ 
 * **SUCCESS** _SciPy-bundle-2022.05-intel-2022a.eb_ 
 * **SUCCESS** _Python-2.7.18-GCCcore-11.2.0.eb_ 
 * **SUCCESS** _hypothesis-4.57.1-GCCcore-11.2.0-Python-2.7.18.eb_ 
 * **SUCCESS** _pybind11-2.7.1-GCCcore-11.2.0-Python-2.7.18.eb_ 
 * **SUCCESS** _SciPy-bundle-2021.10-foss-2021b-Python-2.7.18.eb_ 

Build succeeded for 14 out of 14 (11 easyconfigs in total)
node098.pri.viking.alces.network - Linux CentOS Linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6138 CPU @ 2.00GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/d1cc0932ecfb0d55b8151f53461eb279 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _SciPy-bundle-2019.10-foss-2019b-Python-3.7.4.eb_ 
 * **SUCCESS** _SciPy-bundle-2019.10-fosscuda-2019b-Python-3.7.4.eb_ 
 * **SUCCESS** _SciPy-bundle-2020.11-foss-2020b.eb_ 
 * **SUCCESS** _SciPy-bundle-2020.11-fosscuda-2020b.eb_ 
 * **SUCCESS** _SciPy-bundle-2021.05-foss-2021a.eb_ 
 * **SUCCESS** _SciPy-bundle-2021.05-intel-2021a.eb_ 
 * **SUCCESS** _SciPy-bundle-2021.10-foss-2021b.eb_ 
 * **SUCCESS** _SciPy-bundle-2021.10-foss-2021b-Python-2.7.18.eb_ 
 * **SUCCESS** _SciPy-bundle-2021.10-intel-2021b.eb_ 
 * **SUCCESS** _SciPy-bundle-2022.05-foss-2022a.eb_ 
 * **SUCCESS** _SciPy-bundle-2022.05-intel-2022a.eb_ 

Build succeeded for 11 out of 11 (11 easyconfigs in total)
node3125.skitty.os - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/fcb67fa6b23e8e779e23ebc5ec3b7402 for a full test report.

**Review by boegel — APPROVED** [review]

**Comment by boegel** [comment]
Thanks a lot for all the effort on this @jfgrimm and @branfosj!

**Pull request merged**